src/main/java/com/liferay/faces/util/product/ProductFactory.java

```diff
@@ -15,61 +15,88 @@
  */
 package com.liferay.faces.util.product;
 
-import java.util.Iterator;
-import java.util.ServiceLoader;
+import javax.faces.FacesWrapper;
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+
+import com.liferay.faces.util.factory.FactoryExtensionFinder;
+import com.liferay.faces.util.product.internal.ProductFactoryImpl;
```

Comment:
@stiemannkj1 I think we were trying to avoid imports of internal classes from API classes. I think the ServiceLoader helped us with that before?